src/lib/y2network/widgets/s390_common.rb

```diff
@@ -244,15 +244,40 @@ def handle(event)
         nil
       end
 
+      # @see CWM::AbstractWidget
       def validate
-        return true if !layer2? || valid_mac?(mac_address_widget.value)
+        return true if !layer2? || !lladdress_for(mac_address_widget.value)
+
+        unless valid_mac?(mac_address_widget.value)
+          report_mac_error
+
+          return false
+        end
+
+        use_selected_mac?
+      end
 
-        report_mac_error
-        false
+      # @see CWM::AbstractWidget
+      def store
+        @settings.layer2 = layer2?
+        @settings.lladdress = layer2? ? lladdress_for(mac_address_widget.value) : nil
       end
 
     private
 
+      def use_selected_mac?
+        Yast::Popup.YesNoHeadline(
+          Yast::Label.WarningMsg,
+          format(
+            # TRANSLATORS: Popup trying to prevent the user to set an specific MAC address
+            _("Specifying a MAC address is optional. \n" \
+            "In most cases letting it empty (default) is the correct choice. \n\n" \
+            "Do you really want to set it it '%s'?"),
```

Comment:
typo: *to set it to*

Additionally, I would add a comma after "In most cases".

Reply:
The wording is a little awkward. Perhaps this:
In most cases, leaving it empty and taking the default assigned by the system is the correct choice
Or:
In most cases, letting it default by leaving this empty is the correct choice

Reply:
Adapted according to your suggestions:

![S390QethMACConfirmation](https://user-images.githubusercontent.com/7056681/114675748-82813a00-9d00-11eb-9c0a-85e41ff633ae.png)